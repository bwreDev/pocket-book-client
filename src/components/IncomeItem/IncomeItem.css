.income-item {
  text-transform: capitalize;
  margin-bottom: 0;
  margin-top: 0;
  display: flex;
  flex-direction: column;
  text-align: center;
}

.income-amount,
.income-description {
  margin: 0;
}

.income-post-time {
  font-size: smaller;
}
