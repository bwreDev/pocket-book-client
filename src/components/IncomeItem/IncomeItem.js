import React from 'react';
import moment from 'moment';
import './IncomeItem.css';

export default function IncomeItem(props) {
  const cleanDate = moment(props.input.date_added).format(
    'MMM Do YY, h:mm a'
  );

  return (
    <div className='income-item'>
      <p className='income-amount'>${props.input.amount}</p>
      <p className='income-description'>{props.input.content}</p>
      <span className='income-post-time'>
        {props.input.date_added ? cleanDate : ''}
      </span>
    </div>
  );
}

IncomeItem.defaultProps = {
  input: {},
};
