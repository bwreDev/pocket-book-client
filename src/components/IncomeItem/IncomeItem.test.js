import React from 'react';
import ReactDOM from 'react-dom';
import IncomeItem from './IncomeItem';
import renderer from 'react-test-renderer';

it('should render without error', () => {
  const div = document.createElement('div');
  ReactDOM.render(<IncomeItem />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders the UI as expected', () => {
  const tree = renderer.create(<IncomeItem />).toJSON();
  expect(tree).toMatchSnapshot();
});
