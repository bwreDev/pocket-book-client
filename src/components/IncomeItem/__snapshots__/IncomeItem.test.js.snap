// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
Array [
  <p
    className="income-amount"
  >
    $
  </p>,
  <p
    className="income-description"
  >
     - 
  </p>,
  <span
    className="income-post-time"
  >
    
  </span>,
]
`;
